aggregator/src/datastore.rs

```diff
@@ -1032,12 +1033,43 @@ impl<C: Clock> Transaction<'_, C> {
             .query_one(
                 &stmt,
                 &[
-                    /* task_id */ &task_id.as_ref(),
+                    /* task_id */ task_id.as_ref(),
                     /* batch_interval */ &SqlInterval::from(batch_interval),
                 ],
             )
             .await?;
+        Ok(row
+            .get::<_, Option<i64>>("count")
+            .unwrap_or_default()
+            .try_into()?)
+    }
 
+    /// Return the number of reports in the provided task & batch, regardless of whether the reports
+    /// have been aggregated or collected. Applies only to fixed-size queries.
+    pub async fn count_client_reports_for_batch_id(
+        &self,
+        task_id: &TaskId,
+        batch_id: &BatchId,
+    ) -> Result<u64, Error> {
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT COUNT(1) AS count FROM report_aggregations
+                JOIN aggregation_jobs ON aggregation_jobs.id = report_aggregations.aggregation_job_id
+                WHERE aggregation_jobs.task_id = (SELECT id FROM tasks WHERE task_id = $1)
+                  AND aggregation_jobs.partial_batch_identifier = $2",
```

Comment:
I think we need a `GROUP BY report_aggregations.client_report_id` to avoid double-counting in the case of nontrivial aggregation parameters.

Reply:
👍🏻 Nice catch. I expanded the test to test for this, as well.